### PR TITLE
Fix empty time_features issue

### PR DESCRIPTION
**Pull request by changebo**
*Issue #367*

*Description of changes:*

* Cause of issue #367: In `AddTimeFeatures`, when `time_features` or `self. date_features` is an empty list, the following snippet would throw an error.
```
        self._full_range_date_features = np.vstack(
            [feat(self.full_date_range) for feat in self.date_features]
        )
```

* After this fix, if `time_features` is empty, `self._full_range_date_features` and `data[self.output_field]` will be set to `None`.

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

**Comment by szha** [comment]
Job PR-387/1 is complete. 
Docs are uploaded to http://gluon-ts-staging.s3-accelerate.dualstack.amazonaws.com/PR-387/1/index.html

**Comment by szha** [comment]
Job PR-387/2 is complete. 
Docs are uploaded to http://gluon-ts-staging.s3-accelerate.dualstack.amazonaws.com/PR-387/2/index.html

**Comment by lostella** [comment]
Looks good, thank you! Could you maybe add a test for this case? You can add a new test module to test/model, or add cases to the test routines in test/model/test_models.py

**Comment by lostella** [comment]
Even better, you can add a case to https://github.com/awslabs/gluon-ts/blob/master/test/test_transform.py

**Comment by changebo** [comment]
Sounds great, will do!

**Comment by szha** [comment]
Job PR-387/3 is complete. 
Docs are uploaded to http://gluon-ts-staging.s3-accelerate.dualstack.amazonaws.com/PR-387/3/index.html

**Comment by changebo** [comment]
I've added a simple test function for the case of empty `time_features`. Please let me know if anything needs to be changed/added. @lostella

**Comment by szha** [comment]
Job PR-387/4 is complete. 
Docs are uploaded to http://gluon-ts-staging.s3-accelerate.dualstack.amazonaws.com/PR-387/4/index.html

File: test/test_transform.py [hunk 1]
```diff
@@ -129,6 +129,26 @@ def test_AddTimeFeatures(start, target, is_train):
...
+        output_field="myout",
+        pred_length=pred_length,
+        time_features=[],
+    )
+
+    assert_serializable(t)
+
+    data = {"start": start, "target": target}
+    res = t.map_transform(data, is_train=is_train)
+    assert not res["myout"]
```
Comment (lostella) [inline test/test_transform.py]: I guess the following check is more appropriate (`not x` is `True` also if `x = []`)

```suggestion
    assert res["myout"] is None
```

**Pull request merged**